lib/src/connection.dart

```diff
@@ -380,15 +380,19 @@ abstract class _PostgreSQLExecutionContextMixin implements PostgreSQLExecutionCo
 
   Future _resolveTableOIDs(List<int> oids) async {
     final unresolvedIDString = oids.join(",");
-    final orderedTableNames =
-        await query("SELECT relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
-
-    final iterator = oids.iterator;
-    orderedTableNames.forEach((tableName) {
-      iterator.moveNext();
-      if (tableName.first != null) {
-        _tableOIDNameMap[iterator.current] = tableName.first;
-      }
+    final orderedTableNames = await query(
+        "SELECT oid, relname FROM pg_class WHERE relkind='r' AND oid IN ($unresolvedIDString) ORDER BY oid ASC");
+
+    final unresolvedOids = oids.toSet();
+    for (List list in orderedTableNames) {
+      final int oid = list[0];
+      final String name = list[1];
+      _tableOIDNameMap[oid] = name;
+      unresolvedOids.remove(oid);
+    }
+    // Make sure that the oid has a key in the map, and won't call resolve next time.
+    unresolvedOids.forEach((oid) {
```

Comment:
Is there any way to have a test case that ensures this works as more queries are made? Forgive me for not being able to recall - what is the purpose of this function overall - something for mapped queries, right?

Reply:
> Forgive me for not being able to recall - what is the purpose of this function overall - something for mapped queries, right?

I believe that is correct.

> Is there any way to have a test case that ensures this works as more queries are made? 

Not sure how to do it easily. I have a use case with cockroachdb, but setting that up seems like... too much work for it.

Reply:
Any idea what oid=0 actually is and why it is being asked for? I assume its some sort of reserved oid, and that it won't matter when generating the results for a mapped query - but I'd like to be sure.

Reply:
Hm. So the way we get the OIDs in the first place is to look at column descriptions - from the database - and they have a reference to the table they belong to's OID. Does this occur in scenarios where the column is derived and not an actual column (e.g. the result of a function)? I want to hone in on the particular issue.

Reply:
I'm not sure of its source, as the current dashboard on my cockroachdb cluster says that it was executed 210000 times. As I have many different kind of queries, and most of them don't add up to 30000, I suspect that it is executed on each result set.